docs/implementing_a_new_project_type.md

```diff
@@ -0,0 +1,51 @@
+Implementing a new project type
+-------------------------------
+
+Implementing a new project type is easy.
+
+- First create a new folder `lib/project_types/<type name>`. This type name
+  will setup how your project is loaded. For instance the if you name your folder
```

Comment:
will setup how your project is loaded. For instance ~~the~~ if you name your folder